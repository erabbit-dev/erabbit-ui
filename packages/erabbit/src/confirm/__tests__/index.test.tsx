import { mount } from '@vue/test-utils'

import { Confirm, showConfirm } from '..'
import { later } from '../../test'

test('should render default Confirm', async () => {
  // @ts-ignore
  const wrapper = mount(
    <Confirm
      title="Title"
      message="This is message"
      visible={true}
      teleport=""
    />,
  )

  await later(0)

  expect(wrapper.html()).toMatchSnapshot()
})

test('should use Confirm component state control confirmation box', async () => {
  const wrapper = mount(
    <Confirm title="Title" message="This is message" teleport="" />,
  )

  await wrapper.setProps({ visible: true })
  await later(0)
  expect(wrapper.find('.er-confirm__wrapper').exists()).toBeTruthy()

  await wrapper.setProps({ visible: false })
  await later(0)
  expect(wrapper.find('.er-confirm__wrapper').exists()).toBeFalsy()

  await wrapper.setProps({ visible: true })
  await later(0)
  await wrapper.find('.er-button--default').trigger('click')
  expect(wrapper.emitted('update:visible')?.[0]).toEqual([false, 'cancel'])
  await wrapper.setProps({ visible: false })
  await later(0)
  expect(wrapper.find('.er-confirm__wrapper').exists()).toBeFalsy()

  await wrapper.setProps({ visible: true })
  await later(0)
  await wrapper.find('.er-icon--close').trigger('click')
  expect(wrapper.emitted('update:visible')?.[1]).toEqual([false, 'cancel'])
  await wrapper.setProps({ visible: false })
  await later(0)
  expect(wrapper.find('.er-confirm__wrapper').exists()).toBeFalsy()

  await wrapper.setProps({ visible: true })
  await later(0)
  await wrapper.find('.er-button--primary').trigger('click')
  expect(wrapper.emitted('update:visible')?.[2]).toEqual([false, 'submit'])
  await wrapper.setProps({ visible: false })
  await later(0)
  expect(wrapper.find('.er-confirm__wrapper').exists()).toBeFalsy()
})

test('should use showConfirm function control confirmation box', async () => {
  // @ts-ignore
  const div = document.createElement('div')
  div.id = 'confirm'

  // @ts-ignore
  document.body.appendChild(div)

  const submit = vi.fn()
  const cancel = vi.fn()

  showConfirm({
    title: 'Title',
    message: 'This is message',
    teleport: '#confirm',
    submit,
    cancel,
  }).catch(() => {})

  await later(0)

  expect(div).toMatchSnapshot()

  expect(div.querySelector('.er-confirm__header h3')?.innerHTML).toBe('Title')
  await (div.querySelector('.er-button--default') as HTMLButtonElement).click()
  expect(cancel).toHaveBeenCalled()

  showConfirm({
    title: '温馨提示',
    message: '这里是提示信息',
    teleport: '#confirm',
    submit,
    cancel,
  }).catch(() => {})

  await later(0)

  expect(div).toMatchSnapshot()
  expect(div.querySelector('.er-confirm__header h3')?.innerHTML).toBe(
    '温馨提示',
  )
  await (div.querySelector('.er-button--primary') as HTMLButtonElement).click()
  expect(submit).toHaveBeenCalled()
})
