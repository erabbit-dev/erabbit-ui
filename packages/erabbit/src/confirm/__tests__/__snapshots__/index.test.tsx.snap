// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render default Confirm 1`] = `
"<transition-stub name=\\"popper\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
  <div class=\\"er-popper er-confirm\\">
    <div class=\\"er-confirm__wrapper active\\">
      <div class=\\"er-confirm__header\\">
        <h3>Title</h3><i class=\\"er-icon er-icon--close\\" style=\\"font-size: 1em; color: rgb(102, 102, 102);\\"></i>
      </div>
      <div class=\\"er-confirm__body\\"><i class=\\"er-icon er-icon--warn\\" style=\\"font-size: 1em;\\"></i>
        <p>This is message</p>
      </div>
      <div class=\\"er-confirm__footer\\"><button class=\\"er-button ellipsis er-button--middle er-button--default\\">
          <!----><span class=\\"er-button__text\\">Cancel</span>
        </button><button class=\\"er-button ellipsis er-button--middle er-button--primary\\">
          <!----><span class=\\"er-button__text\\">OK</span>
        </button></div>
    </div>
  </div>
</transition-stub>"
`;

exports[`should use showConfirm function control confirmation box 1`] = `
<div
  id="confirm"
>
  
  <transition-stub
    appear="false"
    css="true"
    name="popper"
    persisted="false"
  >
    <div
      class="er-popper er-confirm"
    >
      <div
        class="er-confirm__wrapper"
      >
        <div
          class="er-confirm__header"
        >
          <h3>
            Title
          </h3>
          <i
            class="er-icon er-icon--close"
            style="font-size: 1em; color: rgb(102, 102, 102);"
          />
        </div>
        <div
          class="er-confirm__body"
        >
          <i
            class="er-icon er-icon--warn"
            style="font-size: 1em;"
          />
          <p>
            This is message
          </p>
        </div>
        <div
          class="er-confirm__footer"
        >
          <button
            class="er-button ellipsis er-button--middle er-button--default"
          >
            <!---->
            <span
              class="er-button__text"
            >
              
              Cancel
              
            </span>
          </button>
          <button
            class="er-button ellipsis er-button--middle er-button--primary"
          >
            <!---->
            <span
              class="er-button__text"
            >
              
              OK
              
            </span>
          </button>
        </div>
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`should use showConfirm function control confirmation box 2`] = `
<div
  id="confirm"
>
  
  <transition-stub
    appear="false"
    css="true"
    name="popper"
    persisted="false"
  >
    <div
      class="er-popper er-confirm"
    >
      <div
        class="er-confirm__wrapper active"
      >
        <div
          class="er-confirm__header"
        >
          <h3>
            温馨提示
          </h3>
          <i
            class="er-icon er-icon--close"
            style="font-size: 1em; color: rgb(102, 102, 102);"
          />
        </div>
        <div
          class="er-confirm__body"
        >
          <i
            class="er-icon er-icon--warn"
            style="font-size: 1em;"
          />
          <p>
            这里是提示信息
          </p>
        </div>
        <div
          class="er-confirm__footer"
        >
          <button
            class="er-button ellipsis er-button--middle er-button--default"
          >
            <!---->
            <span
              class="er-button__text"
            >
              
              Cancel
              
            </span>
          </button>
          <button
            class="er-button ellipsis er-button--middle er-button--primary"
          >
            <!---->
            <span
              class="er-button__text"
            >
              
              OK
              
            </span>
          </button>
        </div>
      </div>
    </div>
  </transition-stub>
</div>
`;
